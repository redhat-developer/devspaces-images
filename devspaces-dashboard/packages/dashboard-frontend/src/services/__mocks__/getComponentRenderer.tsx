/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

export * from '@testing-library/react';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
