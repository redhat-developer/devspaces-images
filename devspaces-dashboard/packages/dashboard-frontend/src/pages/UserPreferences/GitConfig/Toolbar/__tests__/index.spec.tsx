/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';

import { GitConfigToolbar } from '..';

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnAdd = jest.fn();

describe('GitConfigToolbar', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot();
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  describe('import Git Configuration', () => {
    it('should handle onAdd event', () => {
      renderComponent();

      expect(mockOnAdd).not.toHaveBeenCalled();

      const button = screen.getByRole('button', { name: 'Import Git Configuration' });
      button.click();

      expect(mockOnAdd).toHaveBeenCalled();
    });
  });
});

function getComponent(): React.ReactElement {
  return <GitConfigToolbar onAdd={mockOnAdd} />;
}
