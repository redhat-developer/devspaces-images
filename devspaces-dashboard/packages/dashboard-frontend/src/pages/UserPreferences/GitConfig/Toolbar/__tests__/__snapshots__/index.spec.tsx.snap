// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitConfigToolbar snapshot 1`] = `
<div
  className="pf-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-1"
  data-ouia-component-type="PF4/Toolbar"
  data-ouia-safe={true}
  id="pf-random-id-0"
  style={
    {
      "paddingBottom": 0,
    }
  }
>
  <div
    className="pf-c-toolbar__content"
  >
    <div
      className="pf-c-toolbar__content-section"
    >
      <div
        className="pf-c-toolbar__item pf-m-align-right"
      >
        <button
          aria-disabled={false}
          aria-label={null}
          className="pf-c-button pf-m-link"
          data-ouia-component-id="OUIA-Generated-Button-link-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe={true}
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button"
        >
          <span
            className="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </span>
          Import Git Configuration
        </button>
      </div>
    </div>
    <div
      className="pf-c-toolbar__expandable-content"
      id="pf-random-id-0-expandable-content-0"
    >
      <div
        className="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    className="pf-c-toolbar__content pf-m-hidden"
    hidden={true}
  >
    <div
      className="pf-c-toolbar__group"
    />
  </div>
</div>
`;
